app/controllers/api/v1/projects_controller.rb

```diff
@@ -36,7 +36,35 @@ def create
           end
         end
       end
+
+      def key
+        @project = Project.find(params[:contrib_key][:project_id])
+
+        if @cur_user.id == @project.user_id
+          session[:name] = params[:key_name]
+          session[:key] = params[:key]
+          session[:project_id] = @project.id
+          key = ContribKey.new(contrib_key_params)
+          key.save
+        else
+          errors = 'User does not own the project.'
```

Comment:
Can this error block go in the do | format | below so that we have one less else block? It's checking the same condition below after all.